### PR TITLE
@import and url() link. Fix Microsoft/vscode#29899

### PR DESCRIPTION
**Pull request by octref**

### CODE REVIEW

**Comment by octref** [comment]
I do think we need basic utilities like `url.resolve` or `path.resolve`. We should either consider adding those to `vscode-uri` or write a small module that just does that.

Alternatively, we could include a dependency like https://github.com/medialize/URI.js. So far our support for HTML/CSS/JSON etc are just single file only so this isn't a problem, but once we start to add features that work across files we need such utilities.

File: src/services/cssNavigation.ts [hunk 1]
```diff
@@ -5,14 +5,16 @@
 'use strict';
 
 import * as nodes from '../parser/cssNodes';
+import { resolve as resolveUrl } from 'url';
```
[the remark this answers is not in the document]
Comment (octref) [reply src/services/cssNavigation.ts]: @aeschli That's my whole argument above. We would need a function to resolve URI / path once we are dealing with cross-file features. I would like to have it in either `vscode-uri` or a small module that we control, so we can use it elsewhere too.

[the remark this answers is not in the document]
Comment (aeschli) [reply src/services/cssNavigation.ts]: I don't think this should go to `vscode-uri`. The problem is that resolve is scheme depended. With `file://` uris, you need to to a real fs path, then keep the root (drive letter, unc server...) and then apply resolve.
What we are looking for is a resolve that works for `http://` and `https://` (`URLs`).

Reimplementing is often tricker than you think. You end up reading the URL spec. I'm in favor of providing it through a context, given that we have URL in nodejs and URL in dom, ready to use (and we do the same for html)

**Comment by octref** [comment]
Updated PR.

**Pull request merged**